packages/app/components/Settings/SettingsPage.tsx

```diff
@@ -27,23 +43,72 @@ const VerticalDivider = styled.div`
 export default function SettingsPage({
   defaultPage,
 }: SettingsPageProps): ReactElement {
-  const profile = useProfile();
+  const { data: profile, error, mutate } = useSWR(`api/v1/profile`, async () =>
+    API.profile.index()
+  );
+
   const [currentSettings, setCurrentSettings] = useState(
     defaultPage || SettingsOptions.PROFILE
   );
+  const [uploading, setUploading] = useState(false);
 
   const avatarSize = useWindowWidth() / 10;
 
+  const beforeUpload = (file) => {
+    const isJpgOrPng = file.type === "image/jpeg" || file.type === "image/png";
+
+    if (!isJpgOrPng) {
+      message.error("You can only upload JPGs or PNGs!");
+    }
+
+    const isLt1M = file.size / 1024 / 1024 < 1;
+    if (!isLt1M) {
+      message.error("Image must smaller than 1MB!");
+    }
+    return isJpgOrPng && isLt1M;
+  };
+
+  console.log(error);
+
   return (
     <Row>
       <Col span={4} style={{ textAlign: "center" }}>
         {avatarSize ? (
-          <AvatarWithInitals
-            style={{ marginTop: "60px", marginBottom: "60px" }}
-            name={profile?.name}
-            size={avatarSize}
-            fontSize={avatarSize * (3 / 7)}
-          />
+          <>
+            {uploading && <Spin />}
```

Comment:
This might already be in your TODO's but I noticed that the spin icon renders way up at the top of the sidebar which causes all the buttons to move up right under it. It would be nice if the spinner could show were the avatar was previously. 

![image](https://user-images.githubusercontent.com/33739155/102965051-e78af880-44a1-11eb-9f48-221d105b4c63.png)